presto-cli/src/main/java/com/facebook/presto/cli/TsvPrinter.java

```diff
@@ -55,6 +55,13 @@ public void printRows(List<List<?>> rows, boolean complete)
     public void finish()
             throws IOException
     {
+        flush();
+    }
+
+    @Override
+    public void flush()
+            throws IOException
+    {
         printRows(ImmutableList.<List<?>>of(), true);
```

Comment:
If you look at the javadoc of `OutputStream.flush()` for example [here](https://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html#flush()) you will see the general contract. And as @electrum mentions above `printRows()` also does a flush (although I am not sure whether we still need it as we do `writer.flush()`) so I don't think it breaks the general contract of `flush()`.